packages/frontend/src/context/Post.ts

```diff
@@ -307,6 +307,107 @@ export class Data {
         )
     }
 
+    editPost(
+        postId: string = '',
+        title: string = '',
+        content: string = '',
+        epk: string = ''
+    ) {
+        const i = userContext.allEpks.findIndex((e) => e === epk)
+        const epoch = i / unirepConfig.numEpochKeyNoncePerEpoch + 1
+        const epkNonce = i % unirepConfig.numEpochKeyNoncePerEpoch
+
+        queueContext.addOp(
+            async (updateStatus) => {
+                if (userContext && userContext.userState) {
+                    updateStatus({
+                        title: 'Updating post',
+                        details: 'Generating zk proof...',
+                    })
+                    const { publicSignals, proof } =
+                        await userContext.userState.genVerifyEpochKeyProof(
+                            epkNonce,
+                            epoch
+                        )
+                    updateStatus({
+                        title: 'Updating post',
+                        details: 'Waiting for TX inclusion...',
+                    })
+                    const apiURL = makeURL(`post/edit/${postId}`)
+                    const r = await fetch(apiURL, {
+                        headers: {
+                            'content-type': 'application/json',
+                        },
+                        body: JSON.stringify({
+                            title,
+                            content,
+                            proof,
+                            publicSignals,
+                        }),
+                        method: 'POST',
+                    })
+                    const { transaction, error, post: _post } = await r.json()
+                    if (error) throw error
+                    await queueContext.afterTx(transaction)
+                    const post = this.convertDataToPost(_post)
+                    this.postsById[post.id] = post
+                    this.save()
```

Comment:
I think it should return transaction hash here
```ts
return { id: transaction, transactionId: transaction }
```

and what is the `id` used for?

Reply:
where is the `{ id: transaction, transactionId: transaction }`

Reply:
https://github.com/Unirep/Unirep-Social/blob/9d8473d9e9a1060d61b76b7c3af0c12bfd4761c2/packages/frontend/src/context/Post.ts#L301

Reply:
so I mean the `editPost` function should return  
```ts
    return { id: transaction, transactionId: transaction }
```
as in `publishPost`
but not sure what is the `id` used for in `publishPost`